api/v1alpha2/tags.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"fmt"
+)
+
+// Tags defines a slice of tags.
+type Tags []string
+
+const (
+	// NameDigitalOceanProviderPrefix is the tag prefix for
+	// cluster-api-provider-digitalocean owned components
+	NameDigitalOceanProviderPrefix = "sigs-k8s-io:capdo"
+	// APIServerRoleTagValue describes the value for the apiserver role
+	APIServerRoleTagValue = "apiserver"
+	// NodeRoleTagValue describes the value for the node role
+	NodeRoleTagValue = "node"
+)
+
+// ClusterNameTag generates the tag with prefix `NameDigitalOceanProviderPrefix`
+// for resources associated with a cluster. It will generated tag like `sigs-k8s-io:capdo:{clusterName}`
+func ClusterNameTag(clusterName string) string {
+	return fmt.Sprintf("%s:%s", NameDigitalOceanProviderPrefix, clusterName)
+}
+
+// ClusterNameRoleTag generates the tag with prefix `NameDigitalOceanProviderPrefix` and `RoleValue` as suffix
+// It will generated tag like `sigs-k8s-io:capdo:{clusterName}:{role}`
+func ClusterNameRoleTag(clusterName, role string) string {
+	return fmt.Sprintf("%s:%s:%s", NameDigitalOceanProviderPrefix, clusterName, role)
+}
+
+// NameTagFromName returns DigitalOcean safe name tag from name
+func NameTagFromName(name string) string {
+	return fmt.Sprintf("name:%s", DOSafeName(name))
+}
+
+// BuildTagParams is used to build tags around an aws resource.
```

Comment:
I believe the comment should say DigitalOcean. :slightly_smiling_face:

Reply:
whops, already fix :grin: